src/components/Dialog/DialogForm.tsx

```diff
@@ -124,9 +124,17 @@ export function DialogForm<T, R = void>({
       {...FormProps}
       onSubmit={async (data, form) => {
         const submitAction = (data as any).submitAction;
+        const isCleanFormAndNotCleanSubmit =
+          !sendIfClean && !form.getState().dirty;
+
+        const isCleanFormAndSendIfCleanDoesNotMatchSubmitAction =
+          typeof sendIfClean === 'string' &&
+          sendIfClean !== submitAction &&
+          !form.getState().dirty;
+
         if (
-          (!sendIfClean && !form.getState().dirty) ||
-          (typeof sendIfClean === 'string' && sendIfClean !== submitAction)
```

Comment:
The 2nd check was missing the check for whether or not the form was clean, so even a dirty form was not calling submit.

Reply:
Good catch @frankyma.

Would you mind changing this to a "shouldCheckDirty"-like var and then only check form state dirty once?
Something like this...
```tsx
const shouldDirtyCheck = !sendIfClean || (typeof sendIfClean === 'string' && sendIfClean !== submitAction);
const sendSubmission = shouldDirtyCheck ? form.getState().dirty : false;

if (!sendSubmission) {
  // early return
}
```

Reply:
Checking once makes sense.  I renamed the variables to say exactly what we're checking to make it a little clearer for me.  Let me know if they make sense

Reply:
@glassblowerscat @CarsonF The reason why we want to handle the delete in here instead of an onClick on the delete button is because the error gets caught in here?

Reply:
It still took me a minute to talk myself through it. I struggled with the `skipSubmit`. Inverting skipSubmit could make it cleaner
```tsx
const shouldSubmit = submitCleanForm || form.getState().dirty;

if (!shouldSubmit) {
  // early return
}
```
Since that if statement is an early return it's very common to have that negate the condition to continue. It seems to be easier to think about the condition to continue as well, esp in this case. Just like your `submitCleanForm` is easier to reason about than my `shouldDirtyCheck` (👍 ).

But I'm still ok with the way it is now, if you want to leave it.

---

> The reason why we want to handle the delete in here instead of an onClick on the delete button is because the error gets caught in here?

Yes submit progress & errors get reported through the form instead of having to manage those two (or 3) states independently.

Reply:
I like it, pushed.